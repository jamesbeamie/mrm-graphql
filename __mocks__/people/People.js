const allPeopleReturnData = {
  data: {
    users: {
      users: [
        {
          email: 'michael.mukalo@andela.com',
          name: 'Michael Mukalo',
          picture: 'https://lh4.googleusercontent.com/-nvDSOOB7tuI/AAAAAAAAAAI/AAAAAAAAABA/otjsk3lztyc/photo.jpg?sz=50',
          location: 'Nairobi',
          id: '5',
          roles: [
            {
              role: 'Admin',
              id: '2',
              __typename: 'Role',
            },
          ],
          __typename: 'User',
        },
        {
          email: 'marlone.akidiva@andela.com',
          name: 'Marlone Akidiva',
          picture: 'https://lh4.googleusercontent.com/-2TgU0EHynu8/AAAAAAAAAAI/AAAAAAAAAAc/ML95I5e2k_c/photo.jpg?sz=50',
          location: 'Nairobi',
          id: '6',
          roles: [
            {
              role: 'Admin',
              id: '2',
              __typename: 'Role',
            },
          ],
          __typename: 'User',
        },
        {
          email: 'george.tibetegya@andela.com',
          name: 'George Tibetegya',
          picture: 'https://lh3.googleusercontent.com/-yhiUccQX5FE/AAAAAAAAAAI/AAAAAAAAAAc/fsBnDfr2fu8/photo.jpg?sz=50',
          location: 'Kampala',
          id: '7',
          roles: [
            {
              role: 'Admin',
              id: '2',
              __typename: 'Role',
            },
          ],
          __typename: 'User',
        },
      ],
      __typename: 'Users',
      pages: 2,
      queryTotal: 11,
      hasNext: true,
      hasPrevious: false,
    },
  },
};

export { allPeopleReturnData as default };
