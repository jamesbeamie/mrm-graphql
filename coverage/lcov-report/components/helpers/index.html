<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/helpers</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> components/helpers
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.12% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>16/132</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">6.35% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/63</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">10% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.6% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>16/127</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="Controls.js"><a href="Controls.js.html">Controls.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file low" data-value="ImageUpload.js"><a href="ImageUpload.js.html">ImageUpload.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="11" class="abs low">0/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	</tr>

<tr>
	<td class="file low" data-value="InputValidators.js"><a href="InputValidators.js.html">InputValidators.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="29" class="abs low">0/29</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="32" class="abs low">0/32</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="29" class="abs low">0/29</td>
	</tr>

<tr>
	<td class="file low" data-value="NavLink.js"><a href="NavLink.js.html">NavLink.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="15" class="abs low">0/15</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="12" class="abs low">0/12</td>
	</tr>

<tr>
	<td class="file low" data-value="QueriesHelpers.js"><a href="QueriesHelpers.js.html">QueriesHelpers.js</a></td>
	<td data-value="30.3" class="pic low"><div class="chart"><div class="cover-fill" style="width: 30%;"></div><div class="cover-empty" style="width:70%;"></div></div></td>
	<td data-value="30.3" class="pct low">30.3%</td>
	<td data-value="33" class="abs low">10/33</td>
	<td data-value="44.44" class="pct low">44.44%</td>
	<td data-value="9" class="abs low">4/9</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="5" class="abs low">1/5</td>
	<td data-value="30.3" class="pct low">30.3%</td>
	<td data-value="33" class="abs low">10/33</td>
	</tr>

<tr>
	<td class="file low" data-value="RangePicker.js"><a href="RangePicker.js.html">RangePicker.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="23" class="abs low">0/23</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="12" class="abs low">0/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="23" class="abs low">0/23</td>
	</tr>

<tr>
	<td class="file low" data-value="TableBody.js"><a href="TableBody.js.html">TableBody.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	</tr>

<tr>
	<td class="file medium" data-value="TableHead.js"><a href="TableHead.js.html">TableHead.js</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	</tr>

<tr>
	<td class="file low" data-value="timeConverter.js"><a href="timeConverter.js.html">timeConverter.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Apr 11 2019 19:58:40 GMT+0300 (East Africa Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
