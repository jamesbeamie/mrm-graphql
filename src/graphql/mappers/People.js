/**
 * This file will have functions to format data returned by queries in the backend
 * to fit into the current implementation of components while doing less work.
 *
 * I have done this because of the changing requirements and data layer of our backend
 * We stand a risk of changing our components whenever the backend changes
 *
 * In my opinion, instead of always modifying our components when the backend changes
 * We can just reformat our data before we pass it to the component.
 *
 * By doing this, we remain with the current implementation of components,
 * no change in snapshots and all we need to do is to change the data mapping methods
 */

export const formatPeopleData = person => ({
  id: person.id,
  name: person.name,
  email: person.email,
  location: person.location,
  accessLevel: person.roles[0].role,
  picture: person.picture,
});

export { formatPeopleData as default };
