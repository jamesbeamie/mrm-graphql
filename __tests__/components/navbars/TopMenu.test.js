import React from 'react';
import { mount } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';
import { BrowserRouter } from 'react-router-dom';
import toastr from 'toastr';
import { GET_LOCATION_QUERY, GET_LOCATIONS_QUERY } from '../../../src/graphql/queries/People';

import { TopMenuComponent } from '../../../src/components/navbars/TopMenu';
import notification from '../../../src/utils/notification';
import { getUserLocation, getAllLocationsFromCache } from '../../../src/components/helpers/QueriesHelpers';
import allLocations from '../../../__mocks__/offices/Locations';


jest.mock('../../../src/components/helpers/QueriesHelpers');
jest.mock('../../../src/utils/Cookie');
jest.mock('../../../src/utils/notification');

const userLocation = {
  id: 1, name: 'Lagos',
};
getUserLocation.mockReturnValue(userLocation);

getAllLocationsFromCache.mockReturnValue(allLocations.data.allLocations);

let wrapper;
const history = {
  push: jest.fn(),
};

const mocks = [
  { request: { query: GET_LOCATION_QUERY }, result: { userLocation } },
  { request: { query: GET_LOCATIONS_QUERY }, result: allLocations.data.allLocations },
];

beforeAll(() => {
  wrapper = mount(
    <MockedProvider
      mocks={mocks}
      addTypename={false}
    >
      <BrowserRouter>
        <TopMenuComponent
          history={history}
        />
      </BrowserRouter>
    </MockedProvider>);
});

describe('TopMenu Component', () => {
  it('Should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should update user info', () => {
    const topMenuWrapper = wrapper.children(0).children(0).children(0).children(0);
    topMenuWrapper.instance().updateUserInfo({
      firstName: 'foo',
      lastName: 'bar',
    });

    expect(topMenuWrapper.state('firstName')).toBe('foo');
  });

  it('Should show error if searchng without criteria', () => {
    notification.mockImplementation(() => () => {});
    wrapper.find('Input').find('input').simulate('keydown', {
      keyCode: 13,
      preventDefault: jest.fn(),
    });
    expect(notification).toHaveBeenCalledWith(toastr, 'error', 'Please select search criteria');
  });

  it('Should show options', () => {
    const topMenuWrapper = wrapper.children(0).children(0).children(0).children(0);
    topMenuWrapper.find('Input').find('input').simulate('focus');
    expect(topMenuWrapper.state('showOptions')).toBe(true);
  });

  it('Should handle query input', () => {
    const topMenuWrapper = wrapper.children(0).children(0).children(0).children(0);
    topMenuWrapper.find('Input').find('input').simulate('change', { target: { value: 'foo' } });
    expect(topMenuWrapper.state('query')).toBe('foo');
  });

  it('Should set search criteria', () => {
    const topMenuWrapper = wrapper.children(0).children(0).children(0).children(0);
    const mockFocusFunc = jest.spyOn(topMenuWrapper.instance(), 'setFocus').mockImplementation(() => {});
    topMenuWrapper.find('button').at(0).simulate('click');
    expect(mockFocusFunc).toHaveBeenCalled();
    expect(topMenuWrapper.state('component')).toBe('rooms');
  });

  it('Should handle search', () => {
    const topMenuWrapper = wrapper.children(0).children(0).children(0).children(0);
    topMenuWrapper.find('Input').find('input').simulate('keydown', {
      keyCode: 13,
      preventDefault: jest.fn(),
    });
    expect(history.push).toHaveBeenLastCalledWith({
      pathname: '/setup',
      state: {
        component: 'rooms',
        query: 'foo',
      },
    });
  });
});
