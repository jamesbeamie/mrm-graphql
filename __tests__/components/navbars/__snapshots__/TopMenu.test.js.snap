// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopMenu Component Should render component 1`] = `
<MockedProvider
  addTypename={false}
  mocks={
    Array [
      Object {
        "request": Object {
          "query": Object {
            "definitions": Array [
              Object {
                "directives": Array [],
                "kind": "OperationDefinition",
                "name": undefined,
                "operation": "query",
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [
                        Object {
                          "arguments": Array [],
                          "kind": "Directive",
                          "name": Object {
                            "kind": "Name",
                            "value": "client",
                          },
                        },
                      ],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "userLocation",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "name",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
                "variableDefinitions": Array [],
              },
            ],
            "kind": "Document",
            "loc": Object {
              "end": 68,
              "start": 0,
            },
          },
        },
        "result": Object {
          "userLocation": Object {
            "id": 1,
            "name": "Lagos",
          },
        },
      },
      Object {
        "request": Object {
          "query": undefined,
        },
        "result": Array [
          Object {
            "id": "2",
            "name": "Kampala",
          },
          Object {
            "id": "3",
            "name": "Nairobi",
          },
          Object {
            "id": "1",
            "name": "Lagos",
          },
        ],
      },
    ]
  }
>
  <ApolloProvider
    client={
      ApolloClient {
        "__operations_cache__": Map {},
        "cache": InMemoryCache {
          "addTypename": false,
          "cacheKeyRoot": KeyTrie {
            "weakness": true,
          },
          "config": Object {
            "addTypename": false,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "freezeResults": false,
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
            "executeSubSelectedArray": [Function],
            "freezeResults": false,
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
        "clearStoreCallbacks": Array [],
        "defaultOptions": Object {},
        "disableNetworkFetches": false,
        "link": MockLink {
          "addTypename": false,
          "mockedResponsesByKey": Object {
            "{\\"variables\\":{},\\"query\\":\\"{\\\\n  userLocation @client {\\\\n    id\\\\n    name\\\\n  }\\\\n}\\\\n\\"}": Array [
              Object {
                "request": Object {
                  "query": Object {
                    "definitions": Array [
                      Object {
                        "directives": Array [],
                        "kind": "OperationDefinition",
                        "name": undefined,
                        "operation": "query",
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [
                                Object {
                                  "arguments": Array [],
                                  "kind": "Directive",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "client",
                                  },
                                },
                              ],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "userLocation",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "id",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "name",
                                    },
                                    "selectionSet": undefined,
                                  },
                                ],
                              },
                            },
                          ],
                        },
                        "variableDefinitions": Array [],
                      },
                    ],
                    "kind": "Document",
                    "loc": Object {
                      "end": 68,
                      "start": 0,
                    },
                  },
                },
                "result": Object {
                  "userLocation": Object {
                    "id": 1,
                    "name": "Lagos",
                  },
                },
              },
            ],
            "{\\"variables\\":{}}": Array [
              Object {
                "request": Object {
                  "query": undefined,
                },
                "result": Array [
                  Object {
                    "id": "2",
                    "name": "Kampala",
                  },
                  Object {
                    "id": "3",
                    "name": "Nairobi",
                  },
                  Object {
                    "id": "1",
                    "name": "Lagos",
                  },
                ],
              },
            ],
          },
        },
        "localState": LocalState {
          "cache": InMemoryCache {
            "addTypename": false,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": false,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "client": [Circular],
        },
        "mutate": [Function],
        "query": [Function],
        "queryDeduplication": true,
        "queryManager": QueryManager {
          "assumeImmutableResults": false,
          "clientAwareness": Object {
            "name": undefined,
            "version": undefined,
          },
          "dataStore": DataStore {
            "cache": InMemoryCache {
              "addTypename": false,
              "cacheKeyRoot": KeyTrie {
                "weakness": true,
              },
              "config": Object {
                "addTypename": false,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
          },
          "fetchQueryRejectFns": Map {},
          "idCounter": 1,
          "inFlightLinkObservables": Map {},
          "link": MockLink {
            "addTypename": false,
            "mockedResponsesByKey": Object {
              "{\\"variables\\":{},\\"query\\":\\"{\\\\n  userLocation @client {\\\\n    id\\\\n    name\\\\n  }\\\\n}\\\\n\\"}": Array [
                Object {
                  "request": Object {
                    "query": Object {
                      "definitions": Array [
                        Object {
                          "directives": Array [],
                          "kind": "OperationDefinition",
                          "name": undefined,
                          "operation": "query",
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [
                                  Object {
                                    "arguments": Array [],
                                    "kind": "Directive",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "client",
                                    },
                                  },
                                ],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "userLocation",
                                },
                                "selectionSet": Object {
                                  "kind": "SelectionSet",
                                  "selections": Array [
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "id",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "name",
                                      },
                                      "selectionSet": undefined,
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                          "variableDefinitions": Array [],
                        },
                      ],
                      "kind": "Document",
                      "loc": Object {
                        "end": 68,
                        "start": 0,
                      },
                    },
                  },
                  "result": Object {
                    "userLocation": Object {
                      "id": 1,
                      "name": "Lagos",
                    },
                  },
                },
              ],
              "{\\"variables\\":{}}": Array [
                Object {
                  "request": Object {
                    "query": undefined,
                  },
                  "result": Array [
                    Object {
                      "id": "2",
                      "name": "Kampala",
                    },
                    Object {
                      "id": "3",
                      "name": "Nairobi",
                    },
                    Object {
                      "id": "1",
                      "name": "Lagos",
                    },
                  ],
                },
              ],
            },
          },
          "localState": LocalState {
            "cache": InMemoryCache {
              "addTypename": false,
              "cacheKeyRoot": KeyTrie {
                "weakness": true,
              },
              "config": Object {
                "addTypename": false,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
            "client": [Circular],
          },
          "mutationStore": MutationStore {
            "store": Object {},
          },
          "onBroadcast": [Function],
          "pollingInfoByQueryId": Map {},
          "queries": Map {},
          "queryDeduplication": true,
          "queryStore": QueryStore {
            "store": Object {},
          },
          "ssrMode": false,
          "transformCache": WeakMap {},
        },
        "reFetchObservableQueries": [Function],
        "resetStore": [Function],
        "resetStoreCallbacks": Array [],
        "store": DataStore {
          "cache": InMemoryCache {
            "addTypename": false,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": false,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
        },
        "typeDefs": undefined,
        "version": "2.6.4",
        "watchQuery": [Function],
      }
    }
  >
    <BrowserRouter>
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "blank",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <TopMenuComponent
          changeUserLocation={[Function]}
          history={
            Object {
              "push": [MockFunction],
            }
          }
        >
          <div
            className="top-menu"
          >
            <div
              className="container size-2-8 nav-menu"
            >
              <div
                className="container content-start nav-left"
              >
                <ul
                  className="inline nav-brand"
                >
                  <li
                    className="logo"
                  >
                    <img
                      alt="Logo"
                      src="https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fconverge_logo.svg?alt=media&token=cd6a1244-91bb-4a1b-8d74-696bd42053ac"
                    />
                  </li>
                  <li
                    className="logo-text"
                  >
                    <h1>
                      CONVERGE
                    </h1>
                  </li>
                </ul>
              </div>
              <div
                className="container content-end nav-right"
              >
                <div
                  className="search-box"
                >
                  <div
                    className="container input-container"
                  >
                    <div
                      className="search-box__input-field"
                    >
                      <Input
                        controlsClass=""
                        id="amenity"
                        inputClass="mrm-input default-input"
                        labelClass="add-resource-controls"
                        labelName=" "
                        name="amenity"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder=""
                        type="text"
                        value=""
                      >
                        <div
                          className="add-resource-controls"
                        >
                          <label
                            htmlFor="amenity"
                          >
                             
                            <input
                              className="mrm-input default-input"
                              defaultValue=""
                              id="amenity"
                              name="amenity"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              placeholder=""
                              type="text"
                            />
                          </label>
                        </div>
                      </Input>
                      <img
                        alt="Search icon"
                        className="search-icon"
                        onClick={[Function]}
                        onKeyPress={[Function]}
                        src="https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fsearch_icon.svg?alt=media&token=ef28a273-f3ea-4370-9720-845bae1cd05e"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="container side-nav"
                >
                  <div
                    className="notifications"
                  >
                    <img
                      alt="Notification icon"
                      src="https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fnotifications_icon.svg?alt=media&token=23b8d4c8-1c47-4591-bb55-2a97807675fd"
                    />
                  </div>
                  <div
                    className="location-dropdown analytics-cover"
                  >
                    <div
                      className="btn-right"
                    >
                      <div
                        className="btn-right__location"
                      >
                        <Button
                          classProp="btn-right__location__btn"
                          handleClick={null}
                          isDisabled={false}
                          title="Lagos"
                          type={2}
                        >
                          <button
                            className="btn-right__location__btn btn-secondary"
                            disabled={false}
                            onClick={null}
                          >
                            <span>
                              Lagos
                            </span>
                          </button>
                        </Button>
                      </div>
                    </div>
                  </div>
                  <div
                    className="profile"
                  >
                    <img
                      alt="Profile icon"
                      className="menu-icon"
                    />
                  </div>
                  <div
                    className="profile-name"
                  >
                    <span
                      className="username"
                    >
                      undefined undefined
                    </span>
                    <withRouter(ProfileMenu)>
                      <Route
                        render={[Function]}
                      >
                        <ProfileMenu
                          history={
                            Object {
                              "action": "POP",
                              "block": [Function],
                              "createHref": [Function],
                              "go": [Function],
                              "goBack": [Function],
                              "goForward": [Function],
                              "length": 1,
                              "listen": [Function],
                              "location": Object {
                                "hash": "",
                                "pathname": "blank",
                                "search": "",
                                "state": undefined,
                              },
                              "push": [Function],
                              "replace": [Function],
                            }
                          }
                          location={
                            Object {
                              "hash": "",
                              "pathname": "blank",
                              "search": "",
                              "state": undefined,
                            }
                          }
                          match={
                            Object {
                              "isExact": false,
                              "params": Object {},
                              "path": "/",
                              "url": "/",
                            }
                          }
                        >
                          <Dropdown
                            content={
                              <span
                                className="preference-dropdown-list"
                              >
                                <NavLink
                                  activeClassName="active"
                                  ariaCurrent="true"
                                  className="preference-link"
                                  to="/preference"
                                >
                                  <div
                                    className="preference-div"
                                  >
                                    <li>
                                      Preferences
                                    </li>
                                  </div>
                                  <img
                                    alt="logout"
                                    className="preference-icon"
                                    src="https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fsettings_gears.svg?alt=media&token=f8e21af6-70f7-4c0a-9115-b6013712abbb"
                                    type="image"
                                  />
                                </NavLink>
                                <div
                                  className="preference-border"
                                />
                                <button
                                  className="logout-btn"
                                  onClick={[Function]}
                                  tabIndex={0}
                                >
                                  Log Out
                                </button>
                              </span>
                            }
                            isPreference={true}
                          >
                            <div
                              className="dropdown-caret"
                            >
                              <button
                                onMouseDown={[Function]}
                              >
                                <span>
                                   ⌄ 
                                </span>
                              </button>
                            </div>
                          </Dropdown>
                        </ProfileMenu>
                      </Route>
                    </withRouter(ProfileMenu)>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </TopMenuComponent>
      </Router>
    </BrowserRouter>
  </ApolloProvider>
</MockedProvider>
`;
